LuaMenu/widgets/gui_steam_coop_handler.lua

```diff
@@ -159,7 +159,30 @@ end
 --------------------------------------------------------------------------------
 -- Downloading
 
-local function CheckDownloads(gameName, mapName, DoneFunc, gameList)
+-- outcome example: "105.1.1-1354-g72b2d55 BAR105" -> "engine/105.1.1-1354-g72b2d55 bar"
+local function GetEnginePath(engineVersion)
+	return ("engine/" .. engineVersion:gsub(" BAR105", " bar")):lower() -- maybe there are more special cases to take in mind here for very old demos or future ones!
+end
+
+local function haveEngineVersion(engineVersion)
+	local springExecutable = Platform.osFamily == "Windows" and "spring.exe" or "spring"
+	return VFS.FileExists(GetEnginePath(engineVersion) .. "//" .. springExecutable)
+end
+
+-- outcome example: https://github.com/beyond-all-reason/spring/releases/download/spring_bar_%7BBAR105%7D105.1.1-1354-g72b2d55/spring_bar_.BAR105.105.1.1-1354-g72b2d55_windows-64-minimal-portable.7z
+local function GetEngineDownloadUrl(engineVersion)
+	local sanitizedEngineVersion = WG.Chobby.Configuration:SanitizeEngineVersion(engineVersion)
+	local pureVersion = sanitizedEngineVersion:gsub(" BAR105", "")
+	local baseUrl = "https://github.com/beyond-all-reason/spring/releases/download/"
+	local versionDir = "spring_bar_%7BBAR105%7D" .. pureVersion .. "/"
+	local platform64 = Platform.osFamily:lower() .. "-64"
+	local fileName = "spring_bar_.BAR105." .. pureVersion .. "_" .. platform64 .. "-minimal-portable.7z"
```

Comment:
This already won't work, new engine releases won't be cut from BAR105 branch, but some other dedicated release branches.

See e.g. https://github.com/beyond-all-reason/spring/releases/tag/spring_bar_%7Bmaster%7D105.1.1-2682-g7363dbc that was cut from `master` branch.

Reply:
I've changed it to work with any branch name, including any that have "-" characters like https://github.com/beyond-all-reason/spring/tree/sprunk-patch-7  

but I don't think I can support any branch names that have / characters in the name, as these are already getting sanitized out. For example https://github.com/beyond-all-reason/spring/tree/badosu/allow-pure-modifiers-keysets will not work here if ever built. 

Is that a problem?

Reply:
https://discord.com/channels/549281623154229250/724924957074915358/1310288362623598792

This discussion indicates `rel-YYYY-MM.N` will be the likely format for branch names in the future, which is compatible with the code here now.